### ISSUE TITLE
Nvidia container toolkit build for RHEL 7 https://hub.docker.com/r/nvidia/container-toolkit

### ISSUE THREAD

**Issue by relyt0925** [NVIDIA/gpu-operator#58]
_The template below is mostly useful for bug reports and support questions. Feel free to remove anything which doesn't apply to you and add more information where it makes sense._

### 1. Quick Debug Checklist
- [ ] Are you running on an Ubuntu 18.04 node?
- [x ] Are you running Kubernetes v1.13+?
- [x ] Are you running Docker (>= 18.06) or CRIO (>= 1.13+)?
- [x ] Do you have `i2c_core` and `ipmi_msghandler` loaded on the nodes?
- [ x] Did you apply the CRD (`kubectl describe clusterpolicies --all-namespaces`)

### 1. Issue or feature description
I am noticing when trying to set GPUs up on a RHEL worker that the nvidia-container-toolkit compiles the nvidia-container-cli with a GLIBC version that is not compatible with RHEL 7. This leads to the following errors when deploying either the gpu-operator or the https://github.com/openshift-psap/special-resource-operator (this is an Openshift 4.3 environment).

```
Events:

  Type     Reason     Age        From                    Message

  ----     ------     ----       ----                    -------

  Normal   Scheduled  <unknown>  default-scheduler       Successfully assigned nvidia-gpu/nvidia-gpu-device-plugin-qdfmn to 10.177.155.99

  Warning  Failed     7m55s      kubelet, 10.177.155.99  Error: container create failed: time="2020-04-27T19:53:10-05:00" level=warning msg="signal: killed"

time="2020-04-27T19:53:10-05:00" level=error msg="container_linux.go:349: starting container process caused \"process_linux.go:449: container init caused \\\"process_linux.go:432: running prestart hook 0 caused \\\\\\\"error running hook: exit status 1, stdout: , stderr: /run/nvidia/toolkit/nvidia-container-cli.real: /lib64/libc.so.6: version `GLIBC_2.27' not found (required by /run/nvidia/toolkit/libnvidia-container.so.1)\\\\\\\\n\\\\\\\"\\\"\""

container_linux.go:349: starting container process caused "process_linux.go:449: container init caused \"process_linux.go:432: running prestart hook 0 caused \\\"error running hook: exit status 1, stdout: , stderr: /run/nvidia/toolkit/nvidia-container-cli.real: /lib64/libc.so.6: version `GLIBC_2.27' not found (required by /run/nvidia/toolkit/libnvidia-container.so.1)\\\\n\\\"\""
```

It appears there needs to be a new version built of this image to support RHEL 7: https://hub.docker.com/r/nvidia/container-toolkit/tags. Are there any plans for that? Would you mind pointing me to the upstream repo that actually has the docker files that build those images?

### 2. Steps to reproduce the issue
Deploy GPU operator or SRO https://github.com/openshift-psap/special-resource-operator in an Openshift 4.3 cluster with RHEL 7 workers. Once the prestart hook is set: all new containers will fail with the above error

### 3. Information to [attach](https://help.github.com/articles/file-attachments-on-issues-and-pull-requests/) (optional if deemed irrelevant)

Basically the runtime hook that is created with the runtime-enablement container `nvidia/container-toolkit:1.0.0-alpha.3` has it's nvidia logic compiled with the glibc version that corresponds to Ubuntu 18.04. Do I potentially need a different container for the RHEL 7.8 kernel? I tried `nvidia/container-toolkit:1.0.0-beta.1-ubi8` as well but that want into a crash loop.

**Comment by relyt0925** [comment]
cc @RenaudWasTaken

**Comment by relyt0925** [comment]
This version also goes into a crash loop for Openshift 4.3 RHEL 7
`nvidia/container-toolkit:1.0.2-ubi8`

**Comment by relyt0925** [comment]
this is the ultimate root cause:
https://github.com/NVIDIA/gpu-operator/issues/32
